ringpop.go

```diff
@@ -340,6 +346,28 @@ func (rp *Ringpop) setState(s state) {
 	}
 }
 
+// RegisterSelfEvictHook registers the eviction hooks that need to be executed
+// before and after self eviction from the membership. It will return an error
+// when ringpop has not been bootstrapped or when the hooks cannot be registered
+// with the membership protocol.
+func (rp *Ringpop) RegisterSelfEvictHook(hooks swim.SelfEvictHook) error {
```

Comment:
"when the hooks cannot be registered with the membership protocol" - when would this happen?